kvclientpool.go

```diff
@@ -196,12 +198,37 @@ func (p *kvClientPool) startNewClientLocked() <-chan struct{} {
 
 	// create the goroutine to actually create the client
 	go func() {
-		p.logger.Debug("creating new client")
+		p.logger.Info("creating new client kv client",
+			zap.Any("clientConfig", clientConfig))
+
+		p.lock.Lock()
+		defer p.lock.Unlock()
+
+		connectErrThrottle := 1 * time.Second
+		for {
+			connectWaitPeriod := connectErrThrottle - time.Since(p.connectErrTime)
+			if connectWaitPeriod > 0 {
```

Comment:
If we invert this too if == 0 { break} then we can unnest most of the code and remove the continue call?

Reply:
When dealing with loops, I usually try to keep the 'common path' to be the one that consumes the body, and then have the 'exceptional case' be in an if block, in this case, the standard case would be that we break without doing anything.